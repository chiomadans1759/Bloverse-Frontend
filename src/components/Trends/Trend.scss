@import 'sass-rem';

.trendCard {
  border: 1px solid #d9d9d9 !important;
  width: rem(343px);
  height: 100%;
  height: rem(240px);
  border-radius: 0 !important;
  &>img {
    border-radius: 0 !important;
    height: 100%;
  }
  a{
    color: #fff;
  }
  &--overlay {
    position: relative;
    background-color: linear-gradient(red, yellow);

    &__content {
      position: absolute;
      text-align: justify;
      align-items: baseline;
      bottom:  rem(60px);
      padding-right: rem(16px);
    }
  }
  &--title, &--text {
    font-size: rem(14px);
  }
  &--text {
    margin-top: rem(-8px);
  }
  &--username {
    font-size: rem(14px);
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: center;
    margin-top: rem(8px);
    font-weight: 600;
    .initials {
      height: rem(16px);
      width: rem(16px);
      font-size: rem(10px);
      text-align: center;
      padding: 0;
      padding-top: rem(1.5px);
      padding-bottom: rem(1.5px);
      background-color: #868686;
      border: none;
    }
    .name {
      margin: 0 rem(5px);
    }
  }
}

@media screen and (min-width: 1020px) {
  .trendCard {
    border: 1px solid #d9d9d9 !important;
    width: rem(420px);
    height: 100%;
    height: rem(240px);
    border-radius: 0 !important;
    &>img {
      border-radius: 0 !important;
    }
    &--overlay {
      position: relative;
      background-color: linear-gradient(red, yellow);
  
      &__content {
        position: absolute;
        text-align: justify;
        align-items: baseline;
        bottom:  rem(60px);
        padding-right: rem(16px);
      }
    }
    &--title, &--text {
      font-size: rem(16px);
    }
    &--text {
      margin-top: rem(-8px);
    }
    &--username {
      font-size: rem(14px);
      width: 100%;
      display: flex;
      flex-direction: row;
      justify-content: flex-start;
      align-items: center;
      margin-top: rem(8px);
      .initials {
        height: rem(16px);
        width: rem(16px);
        font-size: rem(10px);
        text-align: center;
        padding: 0;
        padding-top: rem(1.5px);
        padding-bottom: rem(1.5px);
        background-color: #868686;
        border: none;
      }
      .name {
        margin: 0 rem(5px);
      }
    }
  }
}
