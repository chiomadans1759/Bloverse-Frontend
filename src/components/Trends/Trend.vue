<template>
  <main class="card text-white trendCard">
    <img src="@/assets/photo-4.jpg" class="card-img" alt="trend image">
    <div class="card-img-overlay trendCard--overlay">
      <div class="content trendCard--overlay__content">
        <p class="card-title trendCard--title">Category</p>
        <p class="card-text font-weight-bold trendCard--text">
          <router-link to="/posts/demo">Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret</router-link>
        </p>
        <p class="card-text trendCard--username">
          <a class="btn btn-primary initials" href="#" role="button"></a>
          <span class="name">John Doe</span>
        </p>
      </div>
    </div>
  </main>
</template>

<script>
export default {
  name: 'Trend'
}
</script>

<style lang='scss' scoped>
@import './Trend.scss';
</style>
