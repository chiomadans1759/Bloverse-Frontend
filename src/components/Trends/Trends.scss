
@import 'sass-rem';

.bl-trends {
  margin: rem(12px) rem(8px);
  height: rem(321px);
  height: auto;
  padding: 0;
  &--header {
    margin-bottom: rem(16px);
    .title {
      color: #252525;
      font-size: rem(14px);
    }
    .subtitle {
      color: #868686;
      font-size: rem(12px);
      width: 100%;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }
  .VueCarousel-slide:first-child {
    margin-left: 0;
  }
  .VueCarousel-slide {
    margin: auto rem(8px);
  }
}

@media screen and (min-width: 1020px) {
  .bl-trends {
    margin: rem(16px) auto;
    height: rem(321px);
    height: auto;
    padding: 0;
    &--header {
      margin-bottom: rem(16px);
      .title {
        color: #252525;
        font-size: rem(16px);
      }
      .subtitle {
        color: #868686;
        font-size: rem(14px);
        width: 100%;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
    .VueCarousel-slide:first-child {
      margin-left: 0;
    }
    .VueCarousel-slide {
      margin: auto rem(8px);
    }
  }
}
