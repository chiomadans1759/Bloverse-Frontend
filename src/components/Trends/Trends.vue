<template>
  <main class="container-fluid bl-trends">
    <div class="bl-trends--header">
      <p class="font-weight-bold title">Trending</p>
      <p class="subtitle">
        Lorem ipsum dolor sit amet, vel accumsan
      </p>
    </div>
    <carousel
      :perPage = "perPage"
      navigationEnabled
      paginationActiveColor = "#096DD9"
      paginationColor = "#868686"
      :paginationSize = "paginationSize"
      :paginationPadding = "paginationPadding"
      :minSwipeDistance = "minSwipeDistance"
      :navigationNextLabel = "navigationNextLabel"
      :navigationPrevLabel = "navigationPrevLabel"
      :autoplayTimeout="4000"
      :autoplay = "autoplay"
      :loop = "loop"
    >
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
      <slide>
        <trend />
      </slide>
    </carousel>
  </main>
</template>

<script>
import { Carousel, Slide } from 'vue-carousel';
import Trend from './Trend';

export default {
  name: 'Trends',
  components: { Carousel, Slide, Trend },
  data() {
    return {
      perPage: 2,
      paginationSize: 5,
      paginationPadding: 5,
      minSwipeDistance: 0,
      autoplay: true,
      loop: true,
      navigationNextLabel: `
      <span style="font-size: 20px; color: #000000; cursor: pointer;">
        <i class="fal fa-caret-circle-right"></i>
      </span>
      `,
      navigationPrevLabel: `
      <span style="font-size: 20px; color: #000000; cursor: pointer;">
        <i class="fal fa-caret-circle-left"></i>
      </span>
      `,
    }
  }
}
</script>

<style lang='scss' scoped>
@import './Trends.scss';
</style>
