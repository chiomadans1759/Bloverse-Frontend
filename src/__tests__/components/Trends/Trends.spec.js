import { mount, RouterLinkStub  } from '@vue/test-utils'
import Trends from '@/components/Trends/Trends.vue';

describe('Trends Component', () => {
  it('should render correctly', () => {

    const wrapper = mount(Trends, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
