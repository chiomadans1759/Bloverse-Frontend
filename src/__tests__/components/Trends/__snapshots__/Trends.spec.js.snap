// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trends Component should render correctly 1`] = `
<main
  class="container-fluid bl-trends"
>
  <div
    class="bl-trends--header"
  >
    <p
      class="font-weight-bold title"
    >
      Trending
    </p>
     
    <p
      class="subtitle"
    >
      
      Lorem ipsum dolor sit amet, vel accumsan
    
    </p>
  </div>
   
  <section
    class="VueCarousel"
  >
    <div
      class="VueCarousel-wrapper"
    >
      <div
        class="VueCarousel-inner"
        role="listbox"
        style="transform: translate(0px, 0); visibility: hidden; height: auto;"
      >
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
         
        <div
          class="VueCarousel-slide"
          tabindex="-1"
        >
          <main
            class="card text-white trendCard"
          >
            <img
              alt="trend image"
              class="card-img"
              src="@/assets/photo-4.jpg"
            />
             
            <div
              class="card-img-overlay trendCard--overlay"
            >
              <div
                class="content trendCard--overlay__content"
              >
                <p
                  class="card-title trendCard--title"
                >
                  Category
                </p>
                 
                <p
                  class="card-text font-weight-bold trendCard--text"
                >
                  <a>
                    Lorem ipsum dolor sit amet, vel accumsan liberavisse ex, ea nec elaboraret interpret
                  </a>
                </p>
                 
                <p
                  class="card-text trendCard--username"
                >
                  <a
                    class="btn btn-primary initials"
                    href="#"
                    role="button"
                  />
                   
                  <span
                    class="name"
                  >
                    John Doe
                  </span>
                </p>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
     
    <div
      class="VueCarousel-pagination"
      data-v-438fd353=""
      style="display: none;"
    >
      <ul
        class="VueCarousel-dot-container"
        data-v-438fd353=""
        role="tablist"
      />
    </div>
     
    <!---->
  </section>
</main>
`;
